producer/writer/writer.go

```diff
@@ -180,7 +180,7 @@ func (w *writer) process(update interface{}) error {
 		if err = csw.Init(w.initType); err != nil {
 			w.logger.Errorf("could not init consumer service writer for %s: %v", cs.String(), err)
 			multiErr = multiErr.Add(err)
-			csw.Close()
+			csw.Close(doNotWaitForAcks)
```

Comment:
Not sure why the close types are different for these different scenarios, can you clarify?

Reply:
Added some comments to the calling places